integration-tests/src/main/java/rest/HtmxApp.java

```diff
@@ -0,0 +1,103 @@
+package rest;
+
+import java.util.Collection;
+
+import jakarta.inject.Inject;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Pattern;
+import jakarta.ws.rs.DELETE;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.Path;
+
+import org.jboss.resteasy.reactive.RestForm;
+import org.jboss.resteasy.reactive.RestPath;
+
+import io.quarkiverse.renarde.htmx.HxController;
+import io.quarkus.qute.CheckedTemplate;
+import io.quarkus.qute.TemplateInstance;
+import service.ContactService;
+
+public class HtmxApp extends HxController {
+
+    @Inject
+    ContactService contactService;
+
+    @CheckedTemplate
+    static class Templates {
+
+        public static native TemplateInstance index(Collection<ContactService.Contact> contacts);
+
+        public static native TemplateInstance index$list(Collection<ContactService.Contact> contacts);
+
+        public static native TemplateInstance view(ContactService.Contact contact);
+
+        public static native TemplateInstance edit(ContactService.Contact contact);
+
+    }
+
+    @Path("")
+    public TemplateInstance index() {
+        if (isHxRequest()) {
+            return Templates.index$list(contactService.contacts().values());
+        }
+        return Templates.index(contactService.contacts().values());
+    }
+
+    public TemplateInstance view(@RestPath("id") int id) {
+        if (!contactService.contacts().containsKey(id)) {
+            throw new IllegalArgumentException("Invalid id: " + id);
+        }
+        return Templates.view(contactService.contacts().get(id));
+    }
+
+    public TemplateInstance edit(@RestPath("id") int id) {
+        if (!contactService.contacts().containsKey(id)) {
+            throw new IllegalArgumentException("Invalid id: " + id);
+        }
+        return Templates.edit(contactService.contacts().get(id));
+    }
+
+    @PUT
+    public void save(@RestPath int id, @RestForm @NotBlank @Pattern(regexp = "[A-Z][a-z]+") String firstName,
+            @RestForm @NotBlank @Pattern(regexp = "[A-Z]+") String lastName,
+            @RestForm @Email String email) {
+        if (!contactService.contacts().containsKey(id)) {
+            throw new IllegalArgumentException("Invalid id: " + id);
+        }
+
+        if (validationFailed()) {
+            edit(id);
```

Comment:
Is it normal that `edit` does not return only the form for HX requests?

Reply:
Oh, I see, it looks like `view` and `edit` are actually partial views. I think it is confusing to have partial views outside of `tags/`, but I'm not entirely sure how to improve this.